File: app/src/main/java/com/arturogutierrez/openticator/domain/account/add/activity/AddAccountManuallyActivity.java
package com.arturogutierrez.openticator.domain.account.add.activity;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import com.arturogutierrez.openticator.R;
import com.arturogutierrez.openticator.di.HasComponent;
import com.arturogutierrez.openticator.domain.account.add.di.AddAccountComponent;
import com.arturogutierrez.openticator.domain.account.add.di.AddAccountModule;
import com.arturogutierrez.openticator.domain.account.add.di.DaggerAddAccountComponent;
import com.arturogutierrez.openticator.domain.account.add.view.AddAccountFormFragment;
import com.arturogutierrez.openticator.view.activity.BaseActivity;

public class AddAccountManuallyActivity extends BaseActivity
    implements HasComponent<AddAccountComponent> {

  private AddAccountComponent addAccountComponent;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    initializeActivity(savedInstanceState);
  }

  @Override
  protected int getLayoutResource() {
    return R.layout.activity_toolbar;
  }

  @Override
  public AddAccountComponent getComponent() {
    return addAccountComponent;
  }

  private void initializeActivity(Bundle savedInstanceState) {
    if (savedInstanceState == null) {
      configureInjector();
      configureToolbar();
      showAddAccountFormFragment();
    }
  }

  private void configureInjector() {
    addAccountComponent = DaggerAddAccountComponent.builder()
        .applicationComponent(getApplicationComponent())
        .activityModule(getActivityModule())
        .addAccountModule(new AddAccountModule())
        .build();
  }

  private void configureToolbar() {
    ActionBar actionBar = getSupportActionBar();
    if (actionBar != null) {
      actionBar.setTitle(R.string.add_account_manually);
      actionBar.setDisplayHomeAsUpEnabled(true);
    }
  }

  private void showAddAccountFormFragment() {
    AddAccountFormFragment addAccountFormFragment = new AddAccountFormFragment();
    addFragment(R.id.content_frame, addAccountFormFragment);
  }
}

File: app/src/main/kotlin/com/arturogutierrez/openticator/domain/account/add/activity/AddAccountManuallyActivity.kt
package com.arturogutierrez.openticator.domain.account.add.activity

import android.os.Bundle
import com.arturogutierrez.openticator.R
import com.arturogutierrez.openticator.di.HasComponent
import com.arturogutierrez.openticator.domain.account.add.di.AddAccountComponent
import com.arturogutierrez.openticator.domain.account.add.di.AddAccountModule
import com.arturogutierrez.openticator.domain.account.add.di.DaggerAddAccountComponent
import com.arturogutierrez.openticator.domain.account.add.view.AddAccountFormFragment
import com.arturogutierrez.openticator.view.activity.BaseActivity

class AddAccountManuallyActivity : BaseActivity(), HasComponent<AddAccountComponent> {

  override val component by lazy { buildComponent() }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    initializeActivity(savedInstanceState)
  }

  override val layoutResource: Int
    get() = R.layout.activity_toolbar

  private fun initializeActivity(savedInstanceState: Bundle?) {
    configureToolbar()

    if (savedInstanceState == null) {
      showAddAccountFormFragment()
    }
  }

  private fun buildComponent(): AddAccountComponent {
    return DaggerAddAccountComponent.builder()
        .applicationComponent(applicationComponent)
        .activityModule(activityModule)
        .addAccountModule(AddAccountModule())
        .build()
  }

  private fun configureToolbar() {
    val actionBar = supportActionBar ?: return

    with(actionBar) {
      setTitle(R.string.add_account_manually)
      setDisplayHomeAsUpEnabled(true)
    }
  }

  private fun showAddAccountFormFragment() {
    val addAccountFormFragment = AddAccountFormFragment()
    replaceFragment(R.id.content_frame, addAccountFormFragment)
  }
}
